src/pipeline-resolver.cpp

```
@@ -1,50 +1,423 @@
+            if (!pipeline->isVirtual())
+            {
+                patchCommand(m_commandList, command, pipeline);
+                continue;
+            }
+
+            PipelineKey key = {};
+            key.pipeline = pipeline;
+            if (pipeline->m_program->isSpecializable())
+            {
+                if (!specializationArgs)
+                    return SLANG_FAIL;
```

Cppcheck (2.21.0)

[style] The function 'rbegin' is never used.

(unusedFunction)